symbolics computation with PauliStrings

Fe-r-oz opened the pull request:
Fixes #33

This PR introduces `symbolics` as an optional (weak) dependency to support symbolic computation for `PauliString`.

#### Changes:
- Resolves errors related to `base_show`, `cutoff`, and `binary_kernel`
- Adds `symbolic_symplify` for symbolic simplification
- Includes a new tutorial demonstrating symbolic computations with `PauliString`

nicolasloizeau commented:
Thank you. Please check the tests and docs build.
Why does adding XY makes a `im` appear ?
```
julia> o += a,  "ZX"
a ZX
julia> o += b,  "XY"
a ZX
b*im XY
```

Would be great if simplify also remove the zeros :
```
julia> Cs = simplify(C)
0 111
0 Z1Z
0 Y11
0 X1Z
```

Fe-r-oz commented:
Thank you very much! I have removed all the zeros, added tests as well!

Adding `XY` introduces an  `im` because of the Pauli algebra:  `X * Y = iZ `. The internal multiplication logic tracks this phase factor. When constructing operators,  `Y` terms contribute imaginary phases depending on their position and combination with other Pauli terms. So,  `b `,  `XY ` results in a Pauli string with an  `im` phase to ensure algebraic consistency. I hope that sounds good!

Fe-r-oz commented:
Thank you for your suggestions!!. Now` im` does not appear :) Locally, I have tested tests and doctests as well. 

doctests:
```
julia> doctest(PauliStrings)
[ Info: SetupBuildDirectory: setting up build directory.
[ Info: Doctest: running doctests.
[ Info: Skipped ExpandTemplates step (doctest only).
[ Info: Skipped CrossReferences step (doctest only).
[ Info: Skipped CheckDocument step (doctest only).
[ Info: Skipped Populate step (doctest only).
[ Info: Skipped RenderDocument step (doctest only).
Test Summary:          | Pass  Total   Time
Doctests: PauliStrings |    1      1  23.7s
```

> Isn't it possible to update the current binary_kermel and show instead of multiple dispatch ?

It seems that since we are using `Symbolics.Num`, if we updated these methods we might have to add `Symbolics` as a dependency, so dispatching via extension.

Fe-r-oz commented:
Thank you for your suggestions!! Please help review this PR!!
